assets/js/util.js

```diff
@@ -242,8 +243,26 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function apyCall(requestObject, endpoint) {
+    window[requestObject.requestName] = $.ajax({
```

Comment:
Um, okay. Apologies but I think I still don't have it clear. Could you explain your point with the code snippet below: ( or a code snippet to demonstrate your point would be perfect!)

```
var source =  {
        'data': request,
        'success': handleTranslateSuccessResponse,
        'error': translationNotAvailable,
        'requestName': 'textTranslateRequest'
}


var targetObj = {
	    url: config.APY_URL + endpoint,
        beforeSend: ajaxSend,
        requestName: 'requestName',
        complete: function () {
            ajaxComplete();
            window[options.requestName] = undefined;
        },
        data: options.request,
        type: (options.request.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET',
        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
        dataType: 'json',
        success: options.success,
        error: options.error
}

Object.assign(targetObj, source);

callApy(args?);


```